meson.build

```diff
@@ -586,11 +586,12 @@ endif
 
 catch2 = dependency(
     'catch2',
+    version: '>=3.7.1',
     default_options: [ 'tests=false' ],
     fallback: ['catch2', 'catch2_dep'],
     required: get_option('tests'),
 )
-if catch2.found()
+if catch2.found() and get_option('tests').enabled()
```

Comment:
This isn't correct, please see <https://mesonbuild.com/Reference-manual_functions.html#dependency_required>. `-Dtests=disabled` already skips the dependency lookup and sets `catch2` to an empty dependency object.
```
Dependency catch2 skipped: feature tests disabled
```

Also, no reason to enforce the version requirement. Last time I checked the test code did not use anything specific to catch2 version 3.7.1 and worked with version 2.x.

Reply:
Hi @alebastr , very interesting case:
1. Actually by default meson configuration for waybar provides two or even more features with the same name ```tests```. One of belongs to the project itself and others one to their dependencies. In my case ```tests``` feature is provided by the gtk-layer-shell-0 and catch2 . So when catch2 dependency is not met in the system, meson trough dependency fallback provides ```tests=false```
2. When catch2 dependency is presented in the system then ```tests``` project feature is used. In my system by default ```tests=auto``` which leads to test subproject compilation. In this case for some reason linker failes with the undefined references
----
Need to digging deeper to check why linker fails

Reply:
Looks like catch2 issue. I create pure small project with catch2 dependency only and not able to compile it. Will create issue to catch2

Reply:
Appropriate issue is created catchorg/Catch2#2912